lib/virtual-fs/src/vfs_fs.rs

```diff
@@ -0,0 +1,201 @@
+use crate::*;
+use serde::{Deserialize, Serialize};
+use std::io::{Read, Seek, Write};
+use std::path::Path;
+use std::sync::Arc;
+
+pub use crate::host_fs::{Stderr, Stdin, Stdout};
+
+#[derive(Clone)]
+pub struct VfsFileSystem {
+    inner: Arc<dyn vfs::FileSystem>,
+}
+
+impl FileSystem for VfsFileSystem {
```

Comment:
Why do we need a VfsFileSystem?

Reply:
We don't, strictly, but the code is pretty much written. We'll probably delete it before shipping but I'll keep it around until the PR starts being less of draft in case we decide it is  something that we want.

Reply:
Yeah, I was trying to find a use case for it and was not able to find any. Let's keep the file up until merging just in case it results useful